kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -279,13 +280,120 @@ impl<'tcx> GotocCtx<'tcx> {
         }
     }
 
+    /// Create an initializer for a generator struct.
+    fn codegen_rvalue_generator(&mut self, operands: &[Operand<'tcx>], ty: Ty<'tcx>) -> Expr {
+        let layout = self.layout_of(ty);
+        let discriminant_field = match &layout.variants {
+            Variants::Multiple { tag_encoding: TagEncoding::Direct, tag_field, .. } => tag_field,
+            _ => unreachable!("Generators have more than one variant and use direct encoding"),
+        };
+        let overall_t = self.codegen_ty(ty);
+        let direct_fields = overall_t.lookup_field("direct_fields", &self.symbol_table).unwrap();
+        let mut operands_iter = operands.iter();
+        let direct_fields_expr = Expr::struct_expr_with_nondet_fields(
```

Comment:
can you use `struct_expr_from_values` instead?

Reply:
Good call, this is a left-over from the time where I was still trying to figure out what the operands were about. Fixed.